framework/data/ActiveDataProvider.php

```diff
@@ -159,8 +166,13 @@ protected function prepareTotalCount()
         if (!$this->query instanceof QueryInterface) {
             throw new InvalidConfigException('The "query" property must be an instance of a class that implements the QueryInterface e.g. yii\db\Query or its subclasses.');
         }
-        $query = clone $this->query;
-        return (int) $query->limit(-1)->offset(-1)->orderBy([])->count('*', $this->db);
+        $query = (clone $this->query)->limit(-1)->offset(-1)->orderBy([]);
+        $key = md5((string)$query);
```

Comment:
IMHO, query object serialization is unreliable. For example, logically identical queries produce different hashes.
```
X::find()->andWhere(['id' => 1, 'status' => 1]);
X::find()->andWhere(['status' => 1, 'id' => 1]);
```
Can we simply cache the value and add a kind of flush() function for the cases when we need to recalculate the counter?

Reply:
You can do that in a separate PR, here we are only going to revert the changes, now you could test this branch,

Thanks you.

Reply:
Don't we revert to 2.0.49.x state? There is no query cache at all
https://github.com/yiisoft/yii2/blob/2.0.49.x/framework/data/ActiveDataProvider.php#L168

Reply:
Thks, fixed.